processor/tests/retry.spec.ts

```diff
@@ -114,7 +113,7 @@ describe("retry function", function () {
     test().then(() => { done(); }).catch((err) => { done(err); });
   });
 
-  it("should fail if all attempts return an error", function (done) {
+  it("should fail if all attempts return an error", function (done: Mocha.Done): void {
```

Comment:
Why is this annotation needed everywhere? TS should contextually type this function if the typings for mocha are present...

Reply:
There's a tslint error when I used the same tslint.json from the service bus package.  It has these arguments for typedef
```
    "typedef": [
      true,
      "call-signature",
      "parameter",
      "property-declaration",
      "member-variable-declaration"
    ],
```

When I was adding the types I also felt it is unnecessary.  Do you think we should remove this tslint rule instead?

Reply:
Keeping the changes as discussed in standup meeting. In the future we don't want this tslint rule.

Reply:
Another way to handle this case (for the future possibly), is to
- not use `done` at all when there is no async code inside the `it`
- instead of `done`, mark the callback function as `async`